core/src/main/java/org/apache/druid/data/input/impl/CloudConfigProperties.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.data.input.impl;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Preconditions;
+import org.apache.druid.metadata.PasswordProvider;
+
+import javax.annotation.Nullable;
+import java.util.Objects;
+
+public class CloudConfigProperties
+{
+  @JsonCreator
+  public CloudConfigProperties(
+      @JsonProperty("accessKeyId") @Nullable PasswordProvider accessKeyId,
```

Comment:
Do we want to support Azure with this too? If so Azure is a bit different; requires an account id, and secret key. Can this be extended for that easily?

Reply:
Yes. Maybe we can rename the accessKeyId to something more generic. But all the cloud (s3, G, Azure) requires a "user" and a "password" which this should work for all

Reply:
Making a specific configProperties class for each cloudInputSource without sharing inheritance